stl/inc/span

```diff
@@ -0,0 +1,678 @@
+    using value_type             = remove_cv_t<_Ty>;
+    using index_type             = size_t;
+    using difference_type        = ptrdiff_t;
+    using pointer                = _Ty*;
+    using const_pointer          = const _Ty*;
+    using reference              = _Ty&;
+    using const_reference        = const _Ty&;
+    using iterator               = _Span_iterator<_Ty>;
+    using const_iterator         = _Span_iterator<const _Ty>;
+    using reverse_iterator       = _STD reverse_iterator<iterator>;
+    using const_reverse_iterator = _STD reverse_iterator<const_iterator>;
+
+    static constexpr index_type extent = _Extent;
+
+    // [span.cons] Constructors, copy, and assignment
+    template <size_t _Ext = _Extent, enable_if_t<_Ext == 0 || _Ext == dynamic_extent, int> = 0>
+    constexpr span() noexcept {};
+
+    constexpr span(pointer _Ptr, index_type _Count) noexcept // strengthened
+        : _Mybase(_Count), _Mydata(_Ptr) {
+#if _CONTAINER_DEBUG_LEVEL > 0
+        if constexpr (_Extent != dynamic_extent) {
+            _STL_VERIFY(_Count == _Extent, "span construction from invalid range as count != extent");
+        }
+#endif // _CONTAINER_DEBUG_LEVEL > 0
+    }
+
+    constexpr span(pointer _First, pointer _Last) noexcept // strengthened
+        : _Mybase(static_cast<index_type>(_Last - _First)), _Mydata(_First) {
+#if _CONTAINER_DEBUG_LEVEL > 0
+        _STL_VERIFY(_First <= _Last, "span construction from pointers only valid for first <= last");
+        if constexpr (_Extent != dynamic_extent) {
+            _STL_VERIFY(_Last - _First == _Extent, "span construction from invalid range as last - first != extent");
+        }
+#endif // _CONTAINER_DEBUG_LEVEL > 0
+    }
+
+    template <size_t _Size, enable_if_t<_Extent == dynamic_extent || _Extent == _Size, int> = 0>
+    constexpr span(element_type (&_Arr)[_Size]) noexcept : _Mybase(_Size), _Mydata(_Arr) {}
+
+    template <size_t _Size, enable_if_t<_Extent == dynamic_extent || _Extent == _Size, int> = 0>
+    constexpr span(array<value_type, _Size>& _Arr) noexcept : _Mybase(_Size), _Mydata(_Arr.data()) {}
+
+    // clang-format off
+    template <size_t _Size, size_t _Ext = _Extent,
+        enable_if_t<conjunction_v<is_convertible<const value_type*, element_type*>,
+                                  bool_constant<_Ext == dynamic_extent || _Ext == _Size>>, int> = 0>
+    constexpr span(const array<value_type, _Size>& _Arr) noexcept 
+        : _Mybase(_Size), _Mydata(_Arr.data()) {}
+
+    template <class _Cont, enable_if_t<conjunction_v<_Is_span_compatible_container<_Cont, element_type>,
+		                                             bool_constant<_Extent == dynamic_extent>>, int> = 0>
+    constexpr span(_Cont& _Container)
+        : _Mybase(static_cast<index_type>(_STD size(_Container))), _Mydata(_STD data(_Container)) {}
+
+	template <class _Cont, enable_if_t<conjunction_v<_Is_span_compatible_container<const _Cont, element_type>,

```

Comment:
I believe I see a real tab in this `clang-format off` block, and we always avoid tabs now. Does clang-format mess this up *very* badly? In general we like to leave it on except when there's really no other choice. Note that empty comments like `//` can be used to force line wrapping when clang-format doesn't want to behave well.

Reply:
Actuall it is not that bad anymore
```cpp
    template <size_t _Size, size_t _Ext = _Extent,
        enable_if_t<conjunction_v<is_convertible<const value_type*, element_type*>,
                                  bool_constant<_Ext == dynamic_extent || _Ext == _Size>>, int> = 0>
    constexpr span(const array<value_type, _Size>& _Arr) noexcept 
        : _Mybase(_Size), _Mydata(_Arr.data()) {}
```

Is trasformed to 
```cpp
    template <size_t _Size, size_t _Ext = _Extent,
        enable_if_t<conjunction_v<is_convertible<const value_type*, element_type*>,
                        bool_constant<_Ext == dynamic_extent || _Ext == _Size>>,
            int> = 0>
    constexpr span(const array<value_type, _Size>& _Arr) noexcept : _Mybase(_Size), _Mydata(_Arr.data()) {}
```

So I will remove the clang-format off here